### PR TITLE
412 reseting max agents

### PR DESCRIPTION
**Pull request by schroedtert**
Updates the event mechanism to also allow resetting the door usage.

### CODE REVIEW

File: geometry/Crossing.cpp [hunk 1]
```diff
@@ -226,6 +227,17 @@ int Crossing::GetPartialDoorUsage() const
      return _partialDoorUsage;
 }
 
+void Crossing::ResetDoorUsage()
+{
+     _doorUsage = 0;
+
+     if ((_outflowRate >= std::numeric_limits<double>::max())){
```
Comment (chraibi) [inline geometry/Crossing.cpp]: you mean `<`?

Comment (schroedtert) [reply geometry/Crossing.cpp]: No, the door flow should only be closed, if no outflowRate set but a maxDoorUsage

File: Simulation.cpp [hunk 2]
```diff
@@ -358,7 +358,7 @@ void Simulation::UpdateRoutesAndLocations()
 //    }
 
 #pragma omp parallel for shared(pedsToRemove, allRooms)
-     for (long unsigned int p = 0; p < allPeds.size(); ++p) {
+     for (size_t p = 0; p < allPeds.size(); ++p) {
```
Comment (chraibi) [inline Simulation.cpp]: there is a little conflict here with @xuqiancheng. 

Apparently, in windows, this change does not work. See also [here](https://github.com/JuPedSim/jpscore/pull/442/commits/cd0571e5b586ef22e5abfe39abdbaf8d03e33dd3#diff-441e5250dd8ecdf163d639bb5ac99da4L453)

Comment (schroedtert) [reply Simulation.cpp]: So should we stay with int? Or should we wait till #440 is merged and see if this solves this?

Comment (Ozaq) [reply Simulation.cpp]: @chraibi The only reason i can think of why this does not work is if the library type `size_t` is not "visible" i.e. the STD headers on linux / windows differ a bit in what they expose coincidentally. Looking at `Simulation.h` and 'Simulation.cpp` I see that there is no include `std::map` or `std::shared_ptr` so even those types are only coincidentally visible. It is considered essential practice in C++ to include all types a cpp/h file needs explicitly and not rely on other headers to transitively supply the type.

To fix the size_t issue on windows (and in general)  do in simulation.cpp:
```cpp
#include <cstddef>
```

**Pull request merged**